fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/SelectExpression.java

```diff
@@ -542,14 +542,14 @@ public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candid
                     }
 
                     //
-                    // Last chance for unmapped predicates - if there is a placeholder or a tautology on the other side that is still
-                    // unmapped, we can (and should) remove it from the unmapped other set now. The reasoning is that this predicate is
-                    // not filtering, so it does not cause records to be filtered that are not filtered on the query side.
-                    //
-                    unmappedOtherPredicates
-                            .removeIf(queryPredicate -> queryPredicate instanceof Placeholder || queryPredicate.isTautology());
-
-                    if (!unmappedOtherPredicates.isEmpty()) {
+                    // Last chance for unmapped candidate predicates - if there is a placeholder without compile-time range
+                    // or a tautology on the candidate side that is still unmapped, we can (and should) remove it from the
+                    // unmapped other set now. The reasoning is that this predicate is not filtering, so it does not cause
+                    // records to be filtered that are not filtered on the query side.
+                    remainingUnmappedCandidatePredicates
+                            .removeIf(queryPredicate -> queryPredicate.isTautology() || (queryPredicate instanceof ValueWithRanges && ((ValueWithRanges)queryPredicate).getRanges().isEmpty()));
```

Comment:
Didn't you already create a helper for this or something similar?

Reply:
hmm ... I just created a small helper method in `ValueWithRanges`:

```
    public boolean isEmpty() {
        return getRanges().isEmpty();
    }
```

are you suggesting we create a helper function for the entire condition? `queryPredicate -> queryPredicate.isTautology() || (queryPredicate instanceof ValueWithRanges && ((ValueWithRanges)queryPredicate).isEmpty())` ?